src/daemon/Seat.cpp

```diff
@@ -141,4 +142,15 @@ namespace SDDM {
             VirtualTerminal::jumpToVt(*nextVt, true);
         }
     }
+
+    bool Seat::canTTY() {
+        OrgFreedesktopLogin1ManagerInterface manager(Logind::serviceName(), Logind::managerPath(), QDBusConnection::systemBus());
+        auto seatPath = manager.GetSeat(m_name);
+        OrgFreedesktopLogin1SeatInterface sessionIface(Logind::serviceName(), seatPath.value().path(), QDBusConnection::systemBus());
```

Comment:
I'm wondering whether this works as expected if there's no login1. in that case, `manager` is already invalid so `manager.GetSeat` is even more invalid and `seatPath.value().path()` yet more invalid. Does it print a warning at some point or even abort?

Reply:
Is there a way I can quickly test that on recent Ubuntu?

Reply:
You could probably fake it by forcing it to use the session bus instead of the system bus

Reply:
I think I will just change it to match what `displayStopped` does

Reply:
logind has, but not consolekit. `Logind::serviceName()` returns `org.freedesktop.ConsoleKit` if CK(2) is available.

Reply:
You could possibly hardcode `org.freedesktop.login1` here

Reply:
ok, although I wonder about elogind...

Reply:
I feel like hardcoding a check against ConsoleKit might be better than a check for logind.
I can change it if you feel different

Reply:
IMO it should look at `CanTTY` - if it exists, use it, if not, use the fallback

Reply:
How's this? if CanTTY doesn't exist, it doesn't hit the early return, and falls back to the 'manual' check

Reply:
Looks good! Gave it a try locally as well, no warnings or other complaints.